### PR TITLE
106 Search Contacts

### PR DESCRIPTION
**Pull request by will0684**
# Description

Completed story [#106](https://taiga.dts-stn.com/project/pond-knowlege-management-portal/us/106?milestone=19).

Acceptance Criteria:

- User loads the home page and clicks the "Contact" search sub menu, all contacts are listed in alphabetical ascending order(by Contact Name) in the search results section and the search input field is in focus for user input
- User enters key words in search input, on each character entered(key up) search is performed on all contact data fields and matching results are displayed in search results section
- User clicks on a Contact in search results and the Contact detail view is displayed, clicks "Back to Results" and returns to contact search results page

## Test Instructions

1. npm run dev
1. see that it meets acceptance criteria

## Checklist
- [x] Strings use placeholders for translation (No hard coded strings)
- [ ] Unit tests have been added/updated

### CODE REVIEW

File: pages/view/contact/_id.vue [hunk 1]
```diff
@@ -70,11 +70,9 @@
 
       <div class="flex justify-start mb-4">
         <div class="w-3/12 margins">
-          <nuxt-link :to="localePath('/search/contact')">
-            <AppButton custom_style="btn-cancel" data_cypress="formButton">
-              {{ $t('contact.back') }}
-            </AppButton>
-          </nuxt-link>
+          <AppButton @click="goBack" custom_style="btn-cancel" data_cypress="formButton">
```
Comment (MarcoGoC) [inline pages/view/contact/_id.vue:73]: my bad I used the same data_cypress name for both buttons. each should have a different name

Comment (will0684) [reply pages/view/contact/_id.vue:73]: Kk will fix now

**Comment by will0684** [comment]
> code looks good, but seeing a couple of warnings when i run it locally? The search looks and works nicely though.

Dang yes I'll fix those now

**Comment by will0684** [comment]
> This might just be me but I think there is a bug.
> When reloading the page, if you are on engagement and want to switch to contacts, they don't appear. same applies for vice-versa. If you are on contacts and go to engagement, the engagements do not appear. This only applies once on a refresh.

I just fixed the bug by moving the state action call from fetch() to asyncData()

**Pull request merged**